### PR TITLE
DRILL-6882: Handle the cases where RowKeyJoin's left pipeline being called multiple times.

### PR DESCRIPTION
**Pull request by gparai**
@amansinha100 can you please review the PR? Thanks!

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/IteratorValidator.java [hunk 1]
```diff
@@ -36,7 +46,7 @@ public IteratorValidator(PhysicalOperator child) {
 
   @Override
   protected PhysicalOperator getNewWithChild(PhysicalOperator child) {
-    return new IteratorValidator(child);
+    return new IteratorValidator(child, this.isRepeatable);
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/IteratorValidator.java]: Remove the 'this' qualifier .

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java [hunk 2]
```diff
@@ -47,6 +47,8 @@
   /** For logging/debuggability only. */
   private static volatile int instanceCount;
 
+  private final boolean isRepeatable;
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java]: Similar to the previous variable's javadoc comment, pls add one here and just link to the comment in `IteratorValidator`.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java [hunk 3]
```diff
@@ -257,7 +264,9 @@ public IterOutcome next() {
           // NONE is allowed even without seeing a OK_NEW_SCHEMA. Such NONE is called
           // FAST NONE.
           // NONE moves to terminal high-level state.
-          validationState = ValidationState.TERMINAL;
+          if (!isRepeatable) {
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java]: Pls modify the comment's last line to indicate moving to TERMINAL state is conditional.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorInjector.java [hunk 4]
```diff
@@ -25,13 +25,25 @@
...
 import org.apache.drill.exec.physical.config.IteratorValidator;
+import org.apache.drill.exec.physical.config.RowKeyJoinPOP;
 
 import org.apache.drill.shaded.guava.com.google.common.collect.Lists;
 
 public class IteratorValidatorInjector extends
     AbstractPhysicalVisitor<PhysicalOperator, FragmentContext, ExecutionSetupException> {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(IteratorValidatorInjector.class);
 
+  /* This flag when set creates all the validators as repeatable validators*/
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorInjector.java]: Minor code-style: add a space before the comment end-marker `*/`

Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorInjector.java]: Done

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java (hunk 3 above)
Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java]: Done

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java (hunk 2 above)
Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java]: Done

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/IteratorValidator.java (hunk 1 above)
Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/IteratorValidator.java]: Done

**Comment by gparai** [comment]
@amansinha100 I have addressed your review comments. Please take a look.

**Pull request closed without merging**